packages/envd/internal/services/process/handler/cache.go

```diff
@@ -0,0 +1,68 @@
+package handler
+
+import (
+	"sync"
+	"time"
+)
+
+// CachedMultiplex wraps MultiplexedChannel and stores values in TimestampedList
+type CachedMultiplex[T any] struct {
+	forker          ForkSource[T]
+	timestampedList *TimestampedList[T]
+	mu              sync.RWMutex
+}
+
+type ForkSource[T any] interface {
```

Comment:
I could be persuaded this is not necessary, but I think this abstraction makes sense and to me is a basic ReaderWriter interface but for a pub sub system.